include/alpaka/mem/view/Accessor.hpp

```diff
@@ -0,0 +1,195 @@
+/* Copyright 2021 Bernhard Manfred Gruber
+
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS” AND ISC DISCLAIMS ALL WARRANTIES WITH
+ * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL ISC BE LIABLE FOR ANY
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR
+ * IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#pragma once
+
+#include <alpaka/dim/Traits.hpp>
+#include <alpaka/mem/view/Traits.hpp>
+
+namespace alpaka
+{
+    template<typename Pointer, typename Value, typename Idx, typename Dim>
+    struct Accessor;
+
+    template<typename Pointer, typename Value, typename Idx>
+    struct Accessor<Pointer, Value, Idx, DimInt<1>>
+    {
+        auto& operator[](Vec<DimInt<1>, Idx> i) const
+        {
+            return (*this)(i[0]);
+        }
+
+        auto& operator[](Idx i) const
+        {
+            return (*this)(i);
+        }
+
+        auto& operator()(Idx i) const
+        {
+            return p[i];
+        }
+
+        Pointer p;
+        Vec<DimInt<1>, Idx> extents;
+    };
+
+    template<typename Pointer, typename Value, typename Idx>
+    struct Accessor<Pointer, Value, Idx, DimInt<2>>
+    {
+        auto& operator[](Vec<DimInt<2>, Idx> i) const
+        {
+            return (*this)(i[0], i[1]);
+        }
+
+        auto& operator()(Idx y, Idx x) const
+        {
+            return *(reinterpret_cast<Pointer>(reinterpret_cast<char*>(p) + y * rowPitchInBytes) + x);
+        }
+
+        Pointer p;
+        Idx rowPitchInBytes;
+        Vec<DimInt<2>, Idx> extents;
+    };
+
+    template<typename Pointer, typename Value, typename Idx>
+    struct Accessor<Pointer, Value, Idx, DimInt<3>>
+    {
+        auto& operator[](Vec<DimInt<3>, Idx> i) const
+        {
+            return (*this)(i[0], i[1], i[2]);
+        }
+
+        auto& operator()(Idx z, Idx y, Idx x) const
+        {
+            return *(
+                reinterpret_cast<Pointer>(reinterpret_cast<char*>(p) + z * slicePitchInBytes + y * rowPitchInBytes)
+                + x);
+        }
+
+        Pointer p;
+        Idx rowPitchInBytes;
+        Idx slicePitchInBytes;
+        Vec<DimInt<3>, Idx> extents;
+    };
+
+#if 0
+    using Image = cudaTextureObject_t;
+
+    template<typename Value, typename Idx>
+    struct Accessor<Image, Value, Idx, DimInt<1>>
+    {
```

Comment:
We could specialize the `Accessor` with images as `Pointer` types. But first we need Image types/concepts in alpaka.